### PR TITLE
Convert hashicorp/waypoint-helm to GitHub Actions

### PR DESCRIPTION
**Pull request by hc-github-team-es-release-engineering**
Pipeline migrated from [CircleCI](https://app.circleci.com/pipelines/github/hashicorp/waypoint-helm) :tada:

## Manual steps

Please perform the following steps to complete the migration:

### hashicorp/vault-helm/update-helm-charts-index
- [x] Ensure environment variable is updated [here](https://github.com/hashicorp/waypoint-helm/pull/65/files#r1157458574): `HELM_CHARTS_GITHUB_TOKEN`
- [x] Ensure environment variable is updated [here](https://github.com/hashicorp/waypoint-helm/pull/65/files#r1157479581): `SLACK_WEBHOOK`

### CODE REVIEW

**Comment by emilymianeil** [comment]
Dry run without publishing helm charts to verify inputs are correct: https://github.com/hashicorp/waypoint-helm/actions/runs/4547066918

File: .github/workflows/update-helm-charts-index.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          tar -xvzf gh.tar.gz && mkdir -p bin && mv "gh_${version}_linux_amd64/bin/gh" bin/
+      - name: update helm-charts index
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |-
+          ./bin/gh workflow run publish-charts.yml \
+            --repo hashicorp/helm-charts \
+            --ref main \
+            -f SOURCE_TAG="${{ github.ref_name }}" \
+            -f SOURCE_REPO="hashicorp/${{ github.repository }}"
```
Comment (dekimsey) [inline .github/workflows/update-helm-charts-index.yml:44]: Head's up, I believe `github.repository` is fully qualified (`$owner/$name`).

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

File: .github/workflows/update-helm-charts-index.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          git_tag=0.1.18
+          chart_tag=$(yq r Chart.yaml version)
+          if [ "${git_tag}" != "${chart_tag}" ]; then
+            echo "chart version (${chart_tag}) did not match git version (${git_tag})"
+            exit 1
+          fi
+      - name: update helm-charts index
+        env:
+          # FIXME: waypoint team to input token
+          GH_TOKEN: ${{ secrets.HELM_CHARTS_GITHUB_TOKEN }}
```
Comment (emilymianeil) [inline .github/workflows/update-helm-charts-index.yml:35]: We need the `Waypoint` team to generate a token with `actions:write` permissions to the `hashicorp/helm-charts` repo and store it in this repo's (`waypoint-helm`) secrets under the name `HELM_CHARTS_GITHUB_TOKEN`

File: .github/workflows/update-helm-charts-index.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      - name: Send Slack Notification
+        # note: this will skip Slack notification if the job is skipped or cancelled.
+        if: ${{ (success() || failure()) && env.SLACK_MESSAGE != '' }}
+        uses: ./.github/actions/slack-status
+        with:
+          block-color: ${{ env.SLACK_BLOCK_COLOR }}
+          message: ${{ env.SLACK_MESSAGE }}
+          run-url: ${{ env.RUN_URL }}
+          # FIXME: waypoint team to input token
+          slack-webhook-url: ${{ secrets.SLACK_WEBHOOK_URL }}
```
Comment (emilymianeil) [inline .github/workflows/update-helm-charts-index.yml:65]: We need the `Waypoint` team to update this secret in this repository using these guidelines: https://github.com/slackapi/slack-github-action#setup

File: .github/workflows/update-helm-charts-index.yml (hunk 2 above)
Comment (emilymianeil) [reply .github/workflows/update-helm-charts-index.yml:36]: Likely the `hc-github-team-waypoint` PAT since it already has write access to the `hashicorp/helm-charts` repo 😄

Comment (catsby) [reply .github/workflows/update-helm-charts-index.yml:36]: Done!

**Review by catsby — APPROVED** [review]
LGTM, we've added the required secrets

File: .github/workflows/update-helm-charts-index.yml (hunk 3 above)
Comment (catsby) [reply .github/workflows/update-helm-charts-index.yml:65]: Done!

File: .github/workflows/update-helm-charts-index.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+name: hashicorp/waypoint-helm/update-helm-charts-index
```
Comment (alvin-huang) [inline .github/workflows/update-helm-charts-index.yml:1]: Should this be just `update-helm-charts-index` instead?

File: .github/workflows/update-helm-charts-index.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        run: |-
+          echo "Building with Go $(cat ".go-version")"
+          echo "go-version=$(cat ".go-version")" >> "$GITHUB_OUTPUT"
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: ${{ steps.get-go-version.outputs.go-version }}
+      - name: verify Chart version matches tag version
+        run: |-
+          GO111MODULE=on go install github.com/mikefarah/yq/v2@latest
```
Comment (alvin-huang) [inline .github/workflows/update-helm-charts-index.yml:24]: `yq` should be available in the ubuntu-latest image already. Can you try removing this to see if it's still available?

**Review by catsby — APPROVED** [review]
Changes/updates look good to me 👍

File: CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1 @@
+# Submit a helpdesk ticket to add any team other than yours referenced in the CODEOWNERS file -- they must be added to collaborators and teams in the repository settings with maintainer privileges. Remove this file as soon as you have completed this
```
Comment (alvin-huang) [inline CODEOWNERS:1]: Should this contain `@hashicorp/waypoint` like the old file?

File: .github/workflows/update-helm-charts-index.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+          esac
+          SUCCESS_MESSAGE=":tada: The waypoint-helm charts index update succeeded."
+          FAILURE_MESSAGE=":red_circle: The waypoint-helm charts index update failed."
+          echo "success-message=${SUCCESS_MESSAGE}" >> $GITHUB_OUTPUT
+          echo "failure-message=${FAILURE_MESSAGE}" >> $GITHUB_OUTPUT
+      - name: Send Slack Notification
+        # note: this will skip Slack notification if the job is skipped or cancelled.
+        if: ${{ (success() || failure()) }}
+        # FIXME: use new composite actions and ensure all inputs / name align
+        uses: hashicorp/actions-send-slack-status@v1
```
Comment (alvin-huang) [inline .github/workflows/update-helm-charts-index.yml:62]: This seems to be called `https://github.com/hashicorp/action-slack-status` so we should pick one and be consistent, either seems fine to me 😄

File: .github/workflows/update-helm-charts-index.yml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        run: |
+          case "${{ steps.update.conclusion }}" in
+            failure)
+              echo "success-boolean=false" >> $GITHUB_OUTPUT
+              ;;
+            success)
+              echo "success-boolean=true" >> $GITHUB_OUTPUT
+              ;;
+          esac
+          SUCCESS_MESSAGE=":tada: The waypoint-helm charts index update succeeded."
```
Comment (alvin-huang) [inline .github/workflows/update-helm-charts-index.yml:54]: I think this message (and the one below) is slightly misleading. The waypoint-helm chart's index update didn't complete but the trigger to kick off the `publish-charts` workflow on `hashicorp/helm-charts` was successful (if `gh workflow run` didn't return nonzero). 

Maybe rewording this to something like 
```
:tada: waypoint-helm charts index update triggered successfully. View the run at https://github.com/hashicorp/helm-charts/actions/workflows/publish-charts.yml`
```

File: CODEOWNERS (hunk 6 above)
Comment (catsby) [reply CODEOWNERS]: Yep, I changed that in 4c7e02a

File: .github/workflows/update-helm-charts-index.yml (hunk 8 above)
Comment (catsby) [reply .github/workflows/update-helm-charts-index.yml]: That makes sense to me. I updated the success message as you suggested in 4931735. I also modified the failure message, but I wasn't sure on where to direct people so I didn't... suggestions welcome there or we can leave as is.

**Review by catsby — APPROVED** [review]
LGTM

**Review by dekimsey — APPROVED** [review]
Yay!

**Review by dekimsey — APPROVED** [review]
